tests/trustyai/drift/utils.py

```diff
@@ -0,0 +1,244 @@
+TIMEOUT_30SEC: int = 30
+
+
+class TrustyAIServiceRequestHandler:
+    """
+    Class to encapsulate the behaviors associated to the different TrustyAIService requests we make in the tests
+    TODO: It will be moved to a more general file when we start using it in new tests.
+    """
+
+    def __init__(self, token: str, service: TrustyAIService, client: DynamicClient):
+        self.token = token
+        self.service = service
+        self.headers = {"Authorization": f"Bearer {self.token}", "Content-Type": "application/json"}
+        self.service_route = Route(
+            client=client, namespace=service.namespace, name="trustyai-service", ensure_exists=True
+        )
+
+    def _send_request(
+        self,
+        endpoint: str,
+        method: str,
+        data: Optional[Dict[str, Any]] = None,
+        json: Optional[Dict[str, Any]] = None,
+    ) -> Any:
+        url = f"https://{self.service_route.host}{endpoint}"
+
+        if method not in ("GET", "POST"):
+            raise ValueError(f"Unsupported HTTP method: {method}")
+        if method == "GET":
+            return requests.get(url=url, headers=self.headers, verify=False)
+        elif method == "POST":
+            return requests.post(url=url, headers=self.headers, data=data, json=json, verify=False)
+
+    def get_model_metadata(self) -> Any:
+        return self._send_request(endpoint="/info", method="GET")
+
+
+def create_ocp_token(namespace: Namespace) -> str:
+    return subprocess.check_output(["oc", "create", "token", "test-user", "-n", namespace.name]).decode().strip()
+
+
+def send_inference_request(
+    token: str,
+    inference_route: Route,
+    data_batch: Any,
+    file_path: str,
+    max_retries: int = 5,
+) -> None:
+    """
+    Send data batch to inference service with retry logic for network errors.
+
+    Args:
+        token: Authentication token
+        inference_route: Route of the inference service
+        data_batch: Data to be sent
+        file_path: Path to the file being processed
+        max_retries: Maximum number of retry attempts (default: 5)
+
+    Returns:
+        None
+
+    Raises:
+        RequestException: If all retry attempts fail
+    """
+    url: str = f"https://{inference_route.host}{inference_route.instance.spec.path}/infer"
+    headers: Dict[str, str] = {"Authorization": f"Bearer {token}"}
+
+    @retry(
+        stop=stop_after_attempt(max_retries),
+        wait=wait_exponential(multiplier=1, min=4, max=10),
+        retry=retry_if_exception_type(requests.RequestException),
+        before_sleep=lambda retry_state: LOGGER.warning(
+            f"Retry attempt {retry_state.attempt_number} for file {file_path} after error. "
+            f"Waiting {retry_state.next_action.sleep} seconds..."
+        ),
+    )
+    def _make_request() -> None:
+        try:
+            response: requests.Response = requests.post(
+                url=url, headers=headers, data=data_batch, verify=False, timeout=TIMEOUT_30SEC
```

Comment:
Can the verify be set to True to a valid cert here as well?

Reply:
Will be addressed in upcoming PR